google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java

```diff
@@ -166,50 +167,42 @@ private boolean isManagedLibrariesEnabled() {
 
     @Override
     public void afterLibraryAdded(final Library addedLibrary) {
-      AppEngineStandardMavenLibrary library = AppEngineStandardMavenLibrary
-          .getLibraryByMavenDisplayName(addedLibrary.getName());
-
-      if (library == null) {
-        return;
-      }
-
-      final DependencyScope scope = library.getScope();
-
-      new WriteAction() {
-        @Override
-        protected void run(@NotNull Result result) throws Throwable {
-          ModuleRootManager manager = ModuleRootManager.getInstance(context.getModule());
-          ModifiableRootModel model = manager.getModifiableModel();
-
-          model.addLibraryEntry(addedLibrary);
-
-          for (OrderEntry orderEntry : model.getOrderEntries()) {
-            if (orderEntry.getPresentableName().equals(addedLibrary.getName())) {
-              ((ExportableOrderEntry) orderEntry).setScope(scope);
-            }
+      AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(addedLibrary.getName()).ifPresent(
```

Comment:
my opinion on this is that I like the style in general but a couple things could improve readability:

1) the nesting due to the lambda plus the write action etc. is hurting readability. If the contents of the ifPresent lamdba were extracted out to logically named methods I think it would help greatly.
2) Extract out AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(addedLibrary.getName()) to a named variable.